spec.bs

```
@@ -460,28 +467,43 @@ Moreover, each {{SharedStorageWorklet}}'s [=global scopes|list of global scopes=
+
+  Worklets that load cross-origin scripts rely on CORS as a baseline permission mechanism to indicate trusted external origins. However, CORS alone is insufficient for creation of a worklet with cross-origin script whose [=data partition origin=] is the script origin. Unlike simple resource sharing, worklets allow the creator site to execute JavaScript within the context of the target origin. To ensure security, an additional response header, \`<a http-header><code>Shared-Storage-Cross-Origin-Worklet-Allowed</code></a>\`, is required from the script origin.
 
-  Cross-origin worklets rely on CORS as a baseline permission mechanism to indicate trusted external origins. However, CORS alone is insufficient for worklet creation. Unlike simple resource sharing, worklets allow the creator site to execute JavaScript within the context of the target origin. To ensure security, an additional header, \`<a http-header><code>Shared-Storage-Cross-Origin-Worklet-Allowed</code></a>\`, is required.
+  ### Monkey Patch for [=fetch a single module script=] ### {#fetch-a-single-module-script-monkey-patch}
+  The following step will be added to the [=fetch a single module script=] steps, after setting the request's destination (i.e. "9. Set |request|'s [=request/destination=] to the result..."):
+
+  10. If <var ignore=''>destination</var> is "sharedstorageworklet" and <var ignore=''>settingsObject</var>'s [=environment settings object/origin=] is [=opaque origin|opaque=], then [=map/set=] <var ignore=''>moduleMap</var>[(<var ignore=''>url</var>, <var ignore=''>moduleType</var>)] to null, run <var ignore=''>onComplete</var> given null, and abort these steps.
```

I added lines 486-489 to address [your comment](https://github.com/WICG/shared-storage/pull/161/files#r1679841312). Let me know if you think this check resolves the issue. Thanks!

---

I think this isn't necessary: It seems "dataOrigin is not opaque" is already guaranteed by the check in https://pr-preview.s3.amazonaws.com/WICG/shared-storage/pull/161.html#determine-whether-shared-storage-is-allowed-by-context, and if addModule fails early, it won't fetch the script.

---

LGTM

---

Removed the patch for fetch a single module script, since [determine-whether-shared-storage-is-allowed-by-context](https://pr-preview.s3.amazonaws.com/WICG/shared-storage/pull/161.html#determine-whether-shared-storage-is-allowed-by-context) performs the needed check. Thanks, @xyaoinum !